pkg/cmd/roachtest/tests/multitenant_tpch.go

```diff
@@ -71,70 +73,66 @@ func runMultiTenantTPCH(
-	if _, err := singleTenantConn.Exec("SET CLUSTER SETTING kv.range_merge.queue_enabled = false;"); err != nil {
+	if _, err := systemConn.Exec("SET CLUSTER SETTING kv.range_merge.queue_enabled = false;"); err != nil {
 		t.Fatal(err)
 	}
-	runTPCH(singleTenantConn, "" /* url */, 0 /* setupIdx */)
+	runTPCH(gatewayNode, install.SystemInterfaceName, sqlInstance, 0 /* setupIdx */)
 
 	// Restart and wipe the cluster to remove advantage of the second TPCH run.
-	c.Stop(ctx, t.L(), option.DefaultStopOpts())
-	c.Wipe(ctx, secure)
+	c.Wipe(ctx)
 	start()
-	singleTenantConn = c.Conn(ctx, t.L(), 1)
 	// Disable merge queue in the system tenant.
-	if _, err := singleTenantConn.Exec("SET CLUSTER SETTING kv.range_merge.queue_enabled = false;"); err != nil {
+	if _, err := systemConn.Exec("SET CLUSTER SETTING kv.range_merge.queue_enabled = false;"); err != nil {
 		t.Fatal(err)
 	}
 
 	// Now we create a tenant and run all TPCH queries within it.
-	if sharedProcess {
-		db := createInMemoryTenantWithConn(ctx, t, c, appTenantName, c.All(), true /* secure */)
-		defer db.Close()
-		url := fmt.Sprintf("{pgurl:1:%s}", appTenantName)
-		runTPCH(db, url, 1 /* setupIdx */)
-	} else {
-		const (
-			tenantID       = 123
-			tenantHTTPPort = 8081
-			tenantSQLPort  = 30258
-			tenantNode     = 1
-		)
-		_, err := singleTenantConn.Exec(`SELECT crdb_internal.create_tenant($1::INT)`, tenantID)
-		if err != nil {
-			t.Fatal(err)
-		}
-		tenant := createTenantNode(ctx, t, c, c.All(), tenantID, tenantNode, tenantHTTPPort, tenantSQLPort)
-		tenant.start(ctx, t, c, "./cockroach")
-		multiTenantConn, err := gosql.Open("postgres", tenant.pgURL)
-		if err != nil {
-			t.Fatal(err)
-		}
-		// Allow the tenant to be able to split tables. We need to run a dummy
-		// split in order to make sure the capability is propagated before
-		// starting the test, otherwise importing tpch may fail.
-		_, err = singleTenantConn.Exec(
-			`ALTER TENANT [$1] SET CLUSTER SETTING sql.split_at.allow_for_secondary_tenant.enabled=true`, tenantID)
-		if err != nil {
-			t.Fatal(err)
-		}
-		_, err = singleTenantConn.Exec(`ALTER TENANT [$1] GRANT CAPABILITY can_admin_split=true`, tenantID)
-		if err != nil {
-			t.Fatal(err)
-		}
-		testutils.SucceedsSoon(t, func() error {
-			if _, err := multiTenantConn.Exec(`CREATE TABLE IF NOT EXISTS dummysplit (a INT)`); err != nil {
-				return err
-			}
-			_, err = multiTenantConn.Exec(`ALTER TABLE dummysplit SPLIT AT VALUES (0)`)
-			return err
-		})
-		runTPCH(multiTenantConn, "'"+tenant.secureURL()+"'", 1 /* setupIdx */)
+	startOpts := option.DefaultStartSharedVirtualClusterOpts(appTenantName)
+	var separateProcessNode option.NodeListOption
+	if !sharedProcess {
+		separateProcessNode = c.All().RandNode()
+		gatewayNode = separateProcessNode[0]
+		startOpts = option.DefaultStartVirtualClusterOpts(appTenantName, sqlInstance)
+	}
+	c.StartServiceForVirtualCluster(
+		ctx, t.L(), separateProcessNode, startOpts, install.MakeClusterSettings(),
```

Comment:
More of an API nit than a nit for this PR, but it's not immediately obvious to me that `externalNodes` is optional in `c.StartServiceForVirtualCluster`. 

i.e., it's okay\not a mistake here to let `separateProcessNode` be empty since it defaults to starting the storage cluster instead.

Especially since storage cluster is optional, maybe external nodes should be an optional parameter too? Side note, but it also was not obvious to me that not passing in a storage cluster instead defaults to `c.All()`. That might just be a `makeNodes()` detail though.

Reply:
100%. I've been using this API more recently to migrate existing tests and I had a similar feeling. `externalNodes` reflects the `--external-nodes` roachprod API, but in the command line we can choose to not pass that flag, and it changes whether we are creating an external process virtual cluster or not.

In the case of this roachtest API, it gets quite confusing. I was planning to open a quick PR to rename `option.TenantName` to `option.VirtualClusterName` for consistency. I'll also improve this API to make things clearer when I do that.